lib/Dialect/FIRRTL/FIRRTLAttributes.cpp

```diff
@@ -34,26 +34,27 @@ Direction direction::flip(Direction direction) {
   llvm_unreachable("unknown direction");
 }
 
-IntegerAttr direction::packAttribute(MLIRContext *context,
-                                     ArrayRef<Direction> directions) {
+mlir::DenseBoolArrayAttr
+direction::packAttribute(MLIRContext *context, ArrayRef<Direction> directions) {
   // Pack the array of directions into an APInt.  Input is zero, output is one.
-  auto size = directions.size();
-  APInt portDirections(size, 0);
-  for (size_t i = 0; i != size; ++i)
-    if (directions[i] == Direction::Out)
-      portDirections.setBit(i);
-  return IntegerAttr::get(IntegerType::get(context, size), portDirections);
+  SmallVector<bool> dirs;
```

Comment:
Doing a resize (as is done in a later method) would avoid repeated multiple allocations.